### PR TITLE
Update ssh repo clone docs

### PR DESCRIPTION
**Pull request by michaellzc**
## Pull Request approval

Although pull request approval is not enforced for this repository in order to reduce friction, merging without a review will generate a ticket for the docs team to review your changes. So if possible, have your pull request approved before merging.

### CODE REVIEW

**Comment by michaellzc** [comment]
* **#669** <a href="https://app.graphite.dev/github/pr/sourcegraph/docs/669?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @michaellzc and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by loujar — COMMENTED** [review]
I think we should also update the [_Configuration_ sections](https://github.com/sourcegraph/docs/blob/main/docs/admin/code_hosts/github.mdx?plain=1#L383) in each of the individual code host's doc pages

**Comment by MaedahBatool** [comment]
@michaellzc thanks for the PR. I have changed the PR base to our docs working branch. There's some feedback from @loujar. Let me know when it's done. Appreciate your help here. 🙏

File: docs/admin/repo/auth.mdx [hunk 1]
```diff
@@ -4,16 +4,47 @@ If authentication (HTTP(S) or SSH) is required to `git clone` a repository then
...
+  "gitSSHCredential": {
+    // make sure the key is base64 encoded
+    // $ cat ~/.ssh/id_rsa | base64
+    "privateKey": "<base64 encoded of the SSH private key>",
+    "passphrase": "<passphrase if applicable, omit if none is needed>"
+  }
+}
+```
+
+Some providers may require additional configuration, consult the code host specific documentation for more information.
```
Comment (MaedahBatool) [inline docs/admin/repo/auth.mdx:30]: Can we link to the docs here?

File: docs/admin/repo/auth.mdx [hunk 2]
```diff
@@ -4,16 +4,47 @@ If authentication (HTTP(S) or SSH) is required to `git clone` a repository then
...
 - [Sourcegraph with Kubernetes](/admin/deploy/kubernetes/): See [Configure repository cloning via SSH](/admin/deploy/kubernetes/configure#ssh-for-cloning).
 - [Single-container Sourcegraph](/admin/deploy/docker-single-container/): See [the single-container git configuration guide](/admin/deploy/docker-single-container/#git-configuration-and-authentication).
 
->NOTE: Repository access over SSH is not yet supported on [Sourcegraph Cloud](/cloud/).
-
 ## Troubleshooting
 
+### Error: `illegal base64 data at input byte 0`
+
+This error indicates that the provided private key is not base64 encoded. Ensure that the private key is base64 encoded before providing it to Sourcegraph.
```
Comment (MaedahBatool) [inline docs/admin/repo/auth.mdx:42]:
```suggestion
This error indicates that the provided private key is not `base64` encoded. Ensure that the private key is `base64` encoded before providing it to Sourcegraph.
```

**Review by MaedahBatool — CHANGES_REQUESTED** [review]

**Comment by michaellzc** [comment]
> I think we should also update the [_Configuration_ sections](https://github.com/sourcegraph/docs/blob/main/docs/admin/code_hosts/github.mdx?plain=1#L383) in each of the individual code host's doc pages

are you talking about the embedded jsonschema? I believe some automation is responsible to update them during release time.

File: docs/admin/repo/auth.mdx [hunk 3]
```diff
@@ -4,16 +4,47 @@ If authentication (HTTP(S) or SSH) is required to `git clone` a repository then
 
 First, ensure your **Site admin > Manage code hosts** code host configuration is configured to use SSH. For example, by setting the `gitURLType` field to "ssh". Alternatively, you may use the "Generic Git host" code host type, which allows you to directly specify Git repository URLs for cloning.
 
-Then, follow the directions below depending on your deployment type:
+Then, you need to supply the SSH keys and passphrase to Sourcegraph if applicable. You have two options:
+
+- **Option 1**: Configure the SSH credentials in code host configuration in the Web UI.
```
Comment (eseliger) [inline docs/admin/repo/auth.mdx:9]: Should we mention that this is the much preferred option going forward?

**Review by eseliger — APPROVED** [review]

**Pull request merged**